state/validation.go

```diff
@@ -112,6 +112,18 @@ func validateBlock(stateDB dbm.DB, state State, block *types.Block) error {
 		}
 	}
 
+	// Validate block Time
+	if block.Height > 1 {
+		medianTime := MedianTime(block.LastCommit, state.LastValidators)
+		if !block.Time.Equal(medianTime) {
```

Comment:
Should we also require that it's great than the prev block time? It should be if +2/3 are correct, but may be worth the sanity check (we have this commented out above at line 37)